### PR TITLE
Allow smart contract wallets to obtain Proof of Humanity trust bonus

### PR DESCRIPTION
**Pull request by scampVR**
##### Description

Fixes an issue where smart contract wallets can't properly sign the trust bonus verification for proof of humanity. The solution is to attempt to call IsValidSignature() on the wallet address if the usual recoverHash() method fails. I tested specifically on an Argent wallet, but in theory, as long as the wallet supports eip-1271, this should work.

Note: I'm very new to interacting with these tech stacks (python, smart contracts), so the more guidance anyone can provide, the better! Thanks!

- Also includes minor typo fix
##### Refers/Fixes

EIP-1271 https://eips.ethereum.org/EIPS/eip-1271
Argent wallet support for EIP-1271 https://docs.argent.xyz/wallet-connect-and-argent#message-signature

##### Testing

Wasn't able to locate existing tests around similar code, but would be happy to add them if provided some guidance

### CODE REVIEW

File: app/assets/v2/js/pages/profile-trust.js [hunk 1]
```diff
@@ -833,7 +833,7 @@ Vue.component('poh-verify-modal', {
             </div>
             <div v-if="step === 'validation-complete'">
               <div>
-                Your Proof of Humanity verification was successful. Thank you for helping make Gitcoin mroe sybil resistant!
+                Your Proof of Humanity verification was successful. Thank you for helping make Gitcoin more sybil resistant!
```
Comment (schultztimothy) [inline app/assets/v2/js/pages/profile-trust.js:836]: 🧙

File: app/dashboard/poh_utils.py [hunk 2]
```diff
@@ -1,7 +1,18 @@
+import json
+
 POH_CONTRACT_ADDRESS = '0xC5E9dDebb09Cd64DfaCab4011A0D5cEDaf7c9BDb'
 
 _poh_contract = None
 
+EIP_1271_ABI = '[{"constant":true,"inputs":[{"name":"_messageHash","type":"bytes32"},{"name":"_signature","type":"bytes"}],"name":"isValidSignature","outputs":[{"name":"magicValue","type":"bytes4"}],"payable":false,"stateMutability":"view","type":"function"}]'
```
Comment (schultztimothy) [inline app/dashboard/poh_utils.py:7]: Super tiny, but you could change this to the same format as below https://github.com/gitcoinco/web/blob/bbac82ff2bfb8755550f59c1872befac00bfd7ff/app/dashboard/poh_utils.py#L29 and not have to use `json.loads()`

**Review by schultztimothy — APPROVED** [review]
I made one tiny comment but LGTM!

File: app/dashboard/poh_utils.py (hunk 2 above)
Comment (scampVR) [reply app/dashboard/poh_utils.py]: Good point, ty!

**Review by chibie — APPROVED** [review]

**Comment by thelostone-mc** [comment]
@scampVR Yo!  I'll pull this in and test it tom but could we 
- move the ABI to it's own file 
- also to unblock you , could you create a branch of this and throw up the fix on that ? 

Also looks like our CI failed (not due to changes in your PR ) , I'll look into get that running as well

File: app/dashboard/poh_utils.py [hunk 3]
```diff
@@ -2,6 +2,15 @@
 
 _poh_contract = None
 
+EIP_1271_ABI = [{"constant":True,"inputs":[{"name":"_messageHash","type":"bytes32"},{"name":"_signature","type":"bytes"}],"name":"isValidSignature","outputs":[{"name":"magicValue","type":"bytes4"}],"payable":False,"stateMutability":"view","type":"function"}]
```
Comment (thelostone-mc) [inline app/dashboard/poh_utils.py:5]: let's move this to the ABI folder 
https://github.com/gitcoinco/web/tree/master/app/grants/abi

File: app/dashboard/poh_utils.py [hunk 4]
```diff
@@ -2,6 +2,15 @@
...
+
+def is_valid_eip_1271_signature(web3, address, hash, signature) -> bool:
```
Comment (thelostone-mc) [inline app/dashboard/poh_utils.py:7]: I'm assuming this is not restricted to just POH so what do you think about moving this over to https://github.com/gitcoinco/web/blob/master/app/grants/utils.py ?

Comment (scampVR) [reply app/dashboard/poh_utils.py:7]: Perfect, this was going to be my next question! Other CL will need to call it as well, so this is great timing thanks

**Comment by scampVR** [comment]
@thelostone-mc Ok took me a bit to figure out that I needed the empty __init__.py file to be able to import the abi file, but should be working now. Also let me know if the naming on the file is strange and i'll change it to whatever

**Review by thelostone-mc — APPROVED** [review]

**Comment by thelostone-mc** [comment]
This looks good @scampVR  ^_^

**Pull request merged**